added 'other operations' section and certs page (#213)
portworx/pxdocs: content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/other-operations/certs.md

---
title: "Certificates as Kubernetes Secrets"
keywords: certificates, certs, portworx, kubernetes
description: Certificates as Kubernetes secrets.
---

Sometimes it is necessary to store an SSL certificate as a Kubernetes secret. The example here is to secure a third-party S3-compatible objectstore for use with Portworx.

### Creating the secret

>  * Copy your certificate to the Kubernetes master node. We will call it `objectstore.pem` and copy it to a folder `/opt/certs`.

Reviewer: Don't need to copy it to the master node. Just any system that where you can run `kubectl` for that cluster.

Author: Fair point, changed.

>  * Copy your certificate to somewhere `kubectl` is configured for this Kubernetes cluster. We will call it `objectstore.pem` and copy it to a folder `/opt/certs`.

Reviewer 2: ```suggestion
 * Copy your certificate to somewhere `kubectl` is configured for this Kubernetes cluster. We will call the file `objectstore.pem` and copy it to the `/opt/certs` folder.
```

 * Create the secret:

```bash

> $ kubectl -n kube-system create secret generic objectstore-cert --from-file=/opt/certs/

Reviewer 2: Let's make it pastable:

```suggestion
kubectl -n kube-system create secret generic objectstore-cert --from-file=/opt/certs/
```

```

 * Confirm it was created correctly:

```bash

> $ kubectl -n kube-system describe secret objectstore-cert

Reviewer 2: ```suggestion
kubectl -n kube-system describe secret objectstore-cert
```

```

>  * Update the Portworx daemonset to add the mount secret and the environment variable to read the certificate:

Reviewer 2: It's a bit confusing.

What do you think about this?
```suggestion
 * Update the Portworx daemonset to add the mount secret and the environment variable:
```

```bash

> $ kubectl -n kube-system edit ds portworx

Reviewer 2: Let's make it pastable:

```suggestion
kubectl -n kube-system edit ds portworx
```

---
title: "Certificates as Kubernetes Secrets"

> hidden: false

Reviewer 3: By default, it's not hidden. So this line is not required.
